### PR TITLE
Feature/turntable

### PR DESCRIPTION
**Pull request by LordPinguin**

### CODE REVIEW

File: Jackal/Jackal Shared/Playing Field/Shared/GameStates/PirateSelectedState.swift [hunk 1]
```diff
@@ -64,13 +68,24 @@ class PirateSelectedState: TurnState {
...
-        
-        pirate.component(ofType: BoardPositionComponent.self)?.boardPosition = cellPosition
+        if piratePosition.int2Position == cellPosition.int2Position {
+            piratePosition.z += 1
+        } else {
+            piratePosition = cellPosition
+        }
+        piratePositionComponent.boardPosition = piratePosition
         
+        var point: CGPoint
```
Comment (tapoton) [inline Jackal/Jackal Shared/Playing Field/Shared/GameStates/PirateSelectedState.swift]: this still can be constant (`let`)

File: Jackal/Jackal Shared/Playing Field/Shared/GraphConnectors/NodeConnectorDescribing.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import SpriteKit
+
+protocol NodeConnectorDescribing {
+    
+    func createNodes(fieldNode: FieldNodeDescribing, graph: BoardGraph<BoardGraphNode>, x: Int8, y: Int8)
+    
+    func addNodesConnections(fieldNode: FieldNodeDescribing, graph: BoardGraph<BoardGraphNode>, x: Int8, y: Int8)
+    
+    func removeNodesConnections(fieldNode: FieldNodeDescribing, graph: BoardGraph<BoardGraphNode>, x: Int8, y: Int8)
+}
```
Comment (tapoton) [inline Jackal/Jackal Shared/Playing Field/Shared/GraphConnectors/NodeConnectorDescribing.swift:18]: We'll still need all other fields associated with their positions. Some dictionary like `[Position: FieldNodeDescribing]` to determine situations when pirate can't move from land to sea and vice versa.

File: Jackal/Jackal Shared/Playing Field/Shared/GraphConnectors/TurntableNodeConnector.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+//  Jackal
+//
+//  Created by Alexander Savonin on 01/03/2018.
+//  Copyright © 2018 Boring Company. All rights reserved.
+//
+
+struct TurntableNodeConnector: NodeConnectorDescribing {
+    
+    // MARK: Private
+    func castFieldNode(fieldNode: FieldNodeDescribing) -> TurntableNode {
```
Comment (tapoton) [inline Jackal/Jackal Shared/Playing Field/Shared/GraphConnectors/TurntableNodeConnector.swift]: why not `private func` then?

**Review by tapoton — APPROVED** [review]

File: Jackal/Jackal Shared/Playing Field/Shared/GraphConnectors/NodeConnectorDescribing.swift (hunk 2 above)
Comment (azonov) [reply Jackal/Jackal Shared/Playing Field/Shared/GraphConnectors/NodeConnectorDescribing.swift:18]: I'm sure, it is not really necessary to be prepared for it. Better to implement when you will need it :-)

File: Jackal/Jackal Shared/Playing Field/Shared/Level.swift [hunk 4]
```diff
@@ -48,33 +48,25 @@ class Level {
...
-                }
-                
-                let connectedNodes = connectedPositions.flatMap(graph.node(at:))
-                centre.addConnections(to: connectedNodes, bidirectional: false)
+                fieldNode.nodeConnector.addNodesConnections(fieldNode: fieldNode, graph: graph, x: x, y: y)
             }
         }
+        
+        for x in 0..<Int8(size.width) {
+            for y in 0..<Int8(size.height) {
```
Comment (azonov) [inline Jackal/Jackal Shared/Playing Field/Shared/Level.swift:64]: So, now I have a question, why do we use 2 loops?

**Review by azonov — APPROVED** [review]

File: Jackal/Jackal Shared/Playing Field/Shared/GameStates/PirateSelectedState.swift (hunk 1 above)
Comment (LordPinguin) [reply Jackal/Jackal Shared/Playing Field/Shared/GameStates/PirateSelectedState.swift]: Yes, fixed.

File: Jackal/Jackal Shared/Playing Field/Shared/GraphConnectors/TurntableNodeConnector.swift (hunk 3 above)
Comment (LordPinguin) [reply Jackal/Jackal Shared/Playing Field/Shared/GraphConnectors/TurntableNodeConnector.swift]: Yep, fixed.

File: Jackal/Jackal Shared/Playing Field/Shared/Level.swift (hunk 4 above)
Comment (LordPinguin) [reply Jackal/Jackal Shared/Playing Field/Shared/Level.swift:64]: Actually we have 3 iterations:
1) Create nodes
2) Connect all nodes
3) Remove unwanted connections

Comment (tapoton) [reply Jackal/Jackal Shared/Playing Field/Shared/Level.swift:64]: 3 loops

Comment (tapoton) [reply Jackal/Jackal Shared/Playing Field/Shared/Level.swift:64]: First adds nodes
Second adds connections
Third removes connections

**Pull request merged**